cmd/spire-agent/cli/run/run.go

```diff
@@ -124,50 +113,52 @@ func (*RunCLI) Synopsis() string {
 	return "Runs the agent"
 }
 
-func parseFile(filePath string) (*runConfig, error) {
-	c := &runConfig{}
+func parseFile(path string) (*config, error) {
+	c := &config{}
+
+	if path == "" {
+		path = defaultConfigPath
+	}
 
 	// Return a friendly error if the file is missing
-	if _, err := os.Stat(filePath); os.IsNotExist(err) {
-		msg := "could not find config file %s: please use the -config flag"
-		p, err := filepath.Abs(filePath)
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		absPath, err := filepath.Abs(path)
 		if err != nil {
-			p = filePath
-			msg = "could not determine CWD; config file not found at %s: use -config"
+			msg := "could not determine CWD; config file not found at %s: use -config"
+			return nil, fmt.Errorf(msg, path)
 		}
-		return nil, fmt.Errorf(msg, p)
-	}
 
-	data, err := ioutil.ReadFile(filePath)
-	if err != nil {
-		return nil, err
+		msg := "could not find config file %s: please use the -config flag"
+		return nil, fmt.Errorf(msg, absPath)
 	}
-	hclTree, err := hcl.Parse(string(data))
+
+	data, err := ioutil.ReadFile(path)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("unable to read configuration: %v", err)
 	}
-	if err := hcl.DecodeObject(&c, hclTree); err != nil {
-		return nil, err
+
+	if err := hcl.Decode(&c, string(data)); err != nil {
+		return nil, fmt.Errorf("unable to decode configuration: %v", err)
```

Comment:
Could be useful to put the filepath in this error message for debugging.

Reply:
👍 good idea, done